app/src/hid.c

```diff
@@ -75,12 +103,18 @@ int zmk_hid_unregister_mods(zmk_mod_flags modifiers)
 
 int zmk_hid_keypad_press(zmk_key code)
 {
-    if (code >= LCTL && code <= RGUI)
-    {
-        return zmk_hid_register_mod(code - LCTL);
+    zmk_mod_flags mods = SELECT_MODS(code);
+    if (mods) {
+        return zmk_hid_register_mods(mods);
```

Comment:
I could be mis-understanding this code, but this read as "If there is mods, apply them and stop", and would explain the behaviour I'm seeing. The code following this ("If only mods, stop") makes me assume this isn't needed.
```suggestion
        zmk_hid_register_mods(mods);
```